Level-1/dot/dot.cpp

```diff
@@ -0,0 +1,132 @@
+
+    //Initialize rocBLAS error to check the return status of the rocBLAS API functions
+    rocblas_status rstatus = rocblas_status_success;
+
+    //Stride between consecutive values of input vector X (default value is 1)
+    rocblas_int incx = options.incx;
+
+    //Stride between consecutive values of input vector Y (default value is 1)
+    rocblas_int incy = options.incy;
+
+    //Number of elements in input vector X and input vector Y (default value is 5)
+    rocblas_int n = options.n;
+
+    //Edge condition check
+    if(n <= 0)
+    {
+        std::cout << "Value of 'n' should be greater than 0" << std::endl;
+        return 0;
+    }
+
+    //Adjusting the size of input vector X for value of stride (incx) not equal to 1
+    size_t sizeX = (n * incx) >= 0 ? n * incx : -(n * incx);
+
+    //Adjusting the size of input vector Y for value of stride (incy) not equal to 1
+    size_t sizeY = (n * incy) >= 0 ? n * incy : -(n * incy);
+
+    //Allocating memory for the host input vectors X, Y and the host scalar result(To receive the result from the device)
+    std::vector<float> hX(sizeX);
+    std::vector<float> hY(sizeY);
+
+    float hResult = 0.0;
+
+    //Initialising random values to both the host vectors X and Y
+    helpers::fillVectorNormRand<float>(hX);
+    helpers::fillVectorNormRand<float>(hY);
+
+    std::cout << "Input Vectors (X)" << std::endl;
+    helpers::printVector(hX);
+
+    std::cout << "Input Vectors (Y)" << std::endl;
+    helpers::printVector(hY);
+
+    /*Initialising the scalar goldResult, goldResult will be used as a 
+    gold standard to compare our result from rocBLAS SDOT funtion*/
+    float goldResult = 0.0;
+
+    //CPU function for SAXPY
+    for(int i = 0; i < n; i++)
+        goldResult += hX[i * incx] * hY[i * incy];
+
+    //Using rocblas API to create a handle
+    rocblas_handle handle;
+    rstatus = rocblas_create_handle(&handle);
+    CHECK_ROCBLAS_STATUS(rstatus);
+
+    {
+        //Allocating memory for the device vectors X, Y and the scalar Result
+        helpers::DeviceVector<float> dX(sizeX);
+        helpers::DeviceVector<float> dY(sizeY);
+        helpers::DeviceVector<float> dResult(sizeof(float));
+
+        //Tansfer data from host vector X to device vector X
+        herror = hipMemcpy(dX, hX.data(), sizeof(float) * sizeX, hipMemcpyHostToDevice);
+        CHECK_HIP_ERROR(herror);
+
+        //Tansfer data from host vector Y to device vector Y
+        herror = hipMemcpy(dY, hY.data(), sizeof(float) * sizeY, hipMemcpyHostToDevice);
+        CHECK_HIP_ERROR(herror);
+
+        //Asynchronous SDOT calculation on device
+        rstatus = rocblas_sdot(handle, n, dX, incx, dY, incy, dResult);
+
+        CHECK_ROCBLAS_STATUS(rstatus);
+
+        /*Transfer the result from device vector Y to host vector Y, 
+        automatically blocked until result is ready*/
+        herror = hipMemcpy(&hResult, dResult, sizeof(float) * 1, hipMemcpyDeviceToHost);
+

```

Comment:
I would think you should update this simple example to also show the hostPointerMode usage of the function call.   Set the mode and then you can avoid the hipMemcpy afterwards.   You would then explain when the result is available, it will require a synchronization of the stream before you can read the host pointer address.  Can discuss in chat.

Reply:
Thank you for the comment. I have made the corresponding changes.